Frameworks/Foundation/NSOrderedSet.mm

```diff
@@ -14,217 +14,297 @@
 //
 //******************************************************************************
 
-#include "Starboard.h"
+#import <Foundation/NSException.h>
+#import <NSOrderedSetInternal.h>
+#import <Starboard.h>
 #import <StubReturn.h>
+#import <VAListHelper.h>
 
-#include "Foundation/NSOrderedSet.h"
+// Needed to make sure that the returned array behaves like a one way proxy to an array
+@implementation _NSProxyOrderedSetArray
+- (instancetype)initWithOrderedSet:(NSOrderedSet*)orderedSet {
+    if (self = [super init]) {
+        _array = orderedSet->_arrayContainer;
```

Comment:
The proxy array is a reflection of the NSOrdredSet (the internal array), so if the internal array is no longer alive; our retain will cause the _arrayContainer to out live the actual NSOrderedSet